### PR TITLE
fix: all project and keep history bug fix

### PR DESCRIPTION
**Pull request by mathild3r**
- [x] Tests written and linted [ℹ︎](https://github.com/snyk/general/wiki/Tests)
- [ ] Documentation written [ℹ︎](https://github.com/snyk/general/wiki/Documentation)
- [x] Commit history is tidy [ℹ︎](https://github.com/snyk/general/wiki/Git)

### What this does

when running the tool on an all-projects output, if the keepHistory was at false the comment made on the PR were deleted.
ie:
we loop though the projects and run the same algorithm on each so 
if project 1 have an issue a comment was posted but then we re-run for project 2 which was deleting the comment made previously. 

I added a test onto check if this is an all-project run.

### CODE REVIEW

File: src/lib/index.ts [hunk 1]
```diff
@@ -120,17 +120,20 @@ const main = async () => {
             noBaseline,
           );
 
+          let ghPrCommentsCreateResponse: ghPrCommentsStatus
+          const allProjectPost = (i > 0) ? true : false
```
Comment (lili2311) [inline src/lib/index.ts:124]: im very unclear what `allProjectPost` means as a variable 🤔 what does it do?

File: src/lib/index.ts [hunk 2]
```diff
@@ -120,17 +120,20 @@ const main = async () => {
...
+
           if (ghPRNumber)
           {
-            if (snykDeltaResults.result === 0 && !keepHistory) {
+            if (snykDeltaResults.result === 0 && !keepHistory && allProjectPost === false) {
```
Comment (lili2311) [inline src/lib/index.ts]: no need to do the `===` the variable is a boolean already

File: src/lib/index.ts (hunk 1 above)
Comment (mathild3r) [reply src/lib/index.ts:124]: if you don't want to keep the history between runs, we still need to to the delete for project 1, but we don't want to do it for the the next one in the list. So I check the index of the loop if we are at index 2 we are on an all-project output and still in the same run.

**Review by lili2311 — APPROVED** [review]

**Pull request merged**